### PR TITLE
plexamp: init

### PR DESCRIPTION
**Pull request by killercup**
###### Motivation for this change
Brand new rewrite of the standalone Plex music player ([announcment](https://medium.com/plexlabs/plexamp-v3-9af3b10063b4)).
Requires a subscription right now.

~~The app image extraction works and the app starts, but I've not been able to log in. The sign in page opens in Web (even though my default browser is Firefox), which is not able to load any page for some reason. Just wanted to throw this out there for others to test and maybe fix my stupid mistake :)~~

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by afontaine** [comment]
@killercup I wholesale copied this into my home-manager configuration (bumping the version to latest, `3.1.1` at time of writing), and it seems to be working _flawlessly_.

I wonder if there's something in the version bump that changed things?

File: pkgs/applications/audio/plexamp/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ lib, fetchurl, appimageTools, pkgs }:
+
+let
+  pname = "plexamp";
+  version = "3.0.3";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://plexamp.plex.tv/plexamp.plex.tv/desktop/Plexamp-${version}.AppImage";
+    sha256 = "1rh608kcpdx04p2w2nl94b9p96i37sij5shjhgvak041s64wbpg5";
```
Comment (afontaine) [inline pkgs/applications/audio/plexamp/default.nix:10]:
```suggestion
    sha256 = "11hgcysa1x9yqvha6ri4vl1zk8gf8vhcpnh3j38wg9ncd7nz5k2v";
```

Updated `sha256` for version 3.1.1

File: pkgs/applications/audio/plexamp/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ lib, fetchurl, appimageTools, pkgs }:
+
+let
+  pname = "plexamp";
+  version = "3.0.3";
```
Comment (afontaine) [inline pkgs/applications/audio/plexamp/default.nix:5]:
```suggestion
  version = "3.1.1";
```

**Comment by killercup** [comment]
@afontaine thanks! I'm not sure what blocked me from pushing this as a non-draft. Does the desktop file with the icon show up correctly?

**Comment by afontaine** [comment]
I _just_ checked with rofi, and the icon appears to be missing. I also seem to be having intermittent issues building the package as `appimageTools.defaultFhsEnvArgs.multiPkgs pkgs` pulls in [`gst_all_1.gtreamer`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/appimage/default.nix#L85) which is broken.

This seems fixed in #93398 but that isn't merged in yet :(

**Comment by killercup** [comment]
Thanks for checking! I'll follow that issue and see if I can find some time to update this PR when it's fixed.

You're also very welcome to take over if you have some more time :)

**Comment by afontaine** [comment]
good news

@killercup it looks like that PR has been merged :grin: I might be able to find some time evening EDT to check things out as well and see if I can get the icon working.

File: pkgs/applications/audio/plexamp/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  inherit name src;
+
+  multiPkgs = null; # no 32bit needed
+  extraPkgs = pkgs: appimageTools.defaultFhsEnvArgs.multiPkgs pkgs ++ [ pkgs.bash ];
+
+  extraInstallCommands = ''
+    ln -s $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/plexamp.desktop $out/share/applications/plexamp.desktop
+    install -m 444 -D ${appimageContents}/plexamp.png \
+      $out/plexamp.png
```
Comment (afontaine) [inline pkgs/applications/audio/plexamp/default.nix:27]:
```suggestion
      $out/share/icons/hicolor/512x512/apps/plexamp.png
```

**Comment by afontaine** [comment]
@killercup found some  time to play with it earlier than I thought, got the icon working :+1:

**Comment by killercup** [comment]
Awesome, thanks so much, Andrew!

This is now ready to be merged.

**Review by jonringer — COMMENTED** [review]
the app launches fine for me, but I'm unable to do anything after signing in :(

**Comment by afontaine** [comment]
@jonringer were you able to log in or did it not get pasted the initial sign-in? I noticed it took a few attempts to log in, but it also took a few attempts on my Windows installation as well, so I am not sure if it's an issue with nix or with plexamp 🤔

**Comment by jonringer** [comment]
>  but it also took a few attempts on my Windows installation as well, so I am not sure if it's an issue with nix or with plexamp

Ah, it's a feature. Didn't realise :) /s

**Review by jonringer — APPROVED** [review]
diff LGTM
gui launches fine

**Pull request merged**